src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1797,15 +1797,42 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
             $container->setAlias('messenger.default_serializer', $config['serializer']['default_serializer']);
         }
 
+        $failureTransports = [];
+        if ($config['failure_transport']) {
+            if (!isset($config['transports'][$config['failure_transport']])) {
+                throw new LogicException(sprintf('Invalid Messenger configuration: the failure transport "%s" is not a valid transport or service id.', $config['failure_transport']));
+            }
+
+            $container->setAlias('messenger.failure_transports.default', $config['failure_transport']);
+            $failureTransports[] = $config['failure_transport'];
+        }
+        foreach ($config['transports'] as $name => $transport) {
+            if (isset($transport['failure_transport']) && !isset($config['transports'][$transport['failure_transport']])) {
+                throw new LogicException(sprintf('Invalid Messenger configuration: the failure transport "%s" is not a valid transport or service id.', $transport['failure_transport']));
+            }
+            if ($transport['failure_transport']) {
+                $failureTransports[] = $transport['failure_transport'];
+            }
+        }
+
         $senderAliases = [];
         $transportRetryReferences = [];
         foreach ($config['transports'] as $name => $transport) {
             $serializerId = $transport['serializer'] ?? 'messenger.default_serializer';
 
+            $isFailureTransport = false;
+            if (\in_array($name, $failureTransports)) {
+                $isFailureTransport = true;
+            }
+
             $transportDefinition = (new Definition(TransportInterface::class))
                 ->setFactory([new Reference('messenger.transport_factory'), 'createTransport'])
                 ->setArguments([$transport['dsn'], $transport['options'] + ['transport_name' => $name], new Reference($serializerId)])
-                ->addTag('messenger.receiver', ['alias' => $name])
+                ->addTag('messenger.receiver', [
+                        'alias' => $name,
+                        'failure_transport' => $isFailureTransport,
```

Comment:
No `$isFailureTransport` variable needed - we can do:

`failure_transport => \in_array($name, $failureTransports)`